### ISSUE TITLE
Provide means to exit Fullscreen and Presentation modes while using only the mouse (or touchscreen)

### ISSUE THREAD

**Issue by SumatraPeter** [sumatrapdfreader/sumatrapdf#882]
Entering **Fullscreen mode** or **Presentation mode** is easy enough using a mouse and the View menu, but strangely there seems to be _no_ way to exit from these modes without pressing **F11** or **F5** respectively or **Esc** on the keyboard.

Every popular program that offers **Fullscreen mode** (such as browsers or even Windows Explorer) displays the toolbar (or at least the 3 window buttons - minimize/restore/close) momentarily as an overlay when the mouse pointer hits the top of the screen, thus allowing the user to click a button to restore back to the desktop (i.e. non-Fullscreen mode).

If such a solution would require a lot of work, a very easy alternative would be to add an **Exit Fullscreen** command to the context menu only while the app is in Fullscreen mode. Simple and will get the job done with minimum fuss.

---

As for **Presentation mode**, this is slightly more problematic because in this mode no context menu is ever displayed. A left click advances to the next page while a right-click simply goes back a page (the mouse wheel can also be used to navigate). So the simplest solution I can think of is to exit back to the desktop  (i.e. non-Presentation mode) automatically if:

* You're already on the first page and press the RMB (or whatever your secondary mouse button is defined as) to go back a page
* You're already on the first page and scroll the mouse wheel up to go back a page
* You're already on the last page and press the LMB (or whatever your primary mouse button is defined as) to go forward a page
* You're already on the last page and scroll the mouse wheel down to go forward a page

**Note:** The above suggestions will equally help those using a touchscreen instead of a mouse as long as left/right swipe behavior is modified accordingly.

**Comment by GitHubRulesOK** [comment]
Tablets can sometimes have 2 levels of programmable gestures built in.
So mine (with touchpad) accepts 8 compass points and I can readily set one of those quarters such as up-right to exit, without needing to train it. Or in windows10 use a three finger salute to select another task from history.

However, since my tablet is also a touch screen I don't bother as I can easily click 2 fingers together for exit (some need two or more fingers twice, presumably at a pinch to not be confused)

Multi-Mouse buttons could also be programmed on a per app basis to whichever you use most for presentations <kbd>**F5**</kbd> or <kbd>**F11**</kbd> 

In summary a basic 3 button mouse is very much limited to rolling around on the mat.

I agree for feature challenged touch pads or mouses, making a quick exit via the context menu could be the preferred alternative option. So in each mode it only requires the minimal amount of 1 line entry

**Comment by GitHubRulesOK** [comment]
@kjk 
the manual says <kbd>**Shift**</kbd> + F11 = enter full screen mode, 
however, on Windows10   <kbd>**F11**</kbd> alone appears to be a system wide full screen toggle function

Since both modes now have context menus but as Peter points out the F5 "presentation" one is not accessible without a keyboard. Would it be difficult in each screen mode to append a context aware entry to exit on the bottom and for presentation mode perhaps trigger its display with a long or double right click?

**Comment by midenok** [comment]
I don't know why it worked, but I managed to enter/exit fullscreen mode by 2-finger tap. I tried double tap first and it didn't work. Please, provide double tap/click as well as this is commonly used interface of many apps.

**Comment by GitHubRulesOK** [comment]
@midenok
there has generally been nobody working on touch so only other key based current choices are now better than above

For FullScreen (F11) you should be able to hold for screen context then select bottom option, Introduce to anser the first half of this issue.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8eade2b4-4f3b-4f0d-aa44-950db1f18c5c)

For Presentation (F5) usually it would not be used without a presentation pointer (usually programmable) or locked to avoid keyboard less mistakes so without  a Keyboard there are two buttons forwards / back or combined as zoom and setting any of those as escape could be disaster mid showing

Now that escape from presentation is the primary reason this issue is not closed.

**Comment by midenok** [comment]
Why you telling the worse solution against 2-finger tap? Please, don't spend the energy in vain. I'm just saying for addition of 2-finger tap it would be nice to have double click. Nobody is interested in what you currently have.

**Comment by kjk** [comment]
I've made double click with left mouse exit fullscreen / presentation mode.

Hopefully this will also work with touchscreen taps but not tested.

https://github.com/sumatrapdfreader/sumatrapdf/commit/61d7c36b5e659e3666e00f79545aa466e7fedc6b
